ros2/launch.sh.tpl

```diff
@@ -2,10 +2,21 @@
 
 set -o errexit -o nounset -o pipefail
 
-ament_prefix_path="{{ament_prefix_path}}"
-if [ -z "${ament_prefix_path}" ]; then
+if [ -n "{{set_up_ros_home}}" ]; then
+  if [ -z "${ROS_HOME:-}" ] && [ -z "${ROS_LOG_DIR:-}" ]; then
+    ros_output_dir="${TEST_UNDECLARED_OUTPUTS_DIR:-${TEST_TMPDIR:-}}"
+    if [ -n "${ros_output_dir}" ]; then
+      export ROS_HOME="${ros_output_dir}"
+      export ROS_LOG_DIR="${ros_output_dir}"
+    fi
+  fi
+fi
```

Comment:
OK, now I think I understand what you want to do. According to the test encyclopedia BAZEL_TEST is always defined when users do `bazel test`. So, I would do the following

```sh
if BAZEL_TEST defined:
  <reimplement python logic for setting up ROS home and log dirs>
  # NOTE: TEST_TMPDIR is always defined.
  export ROS_HOME=...
  export ROS_LOG_DIR=...
```